src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalUtils.cs

```diff
@@ -18,12 +18,13 @@ public static bool IsBlittableType(TypeDesc type)
                 return false;
             }
 
-            TypeDesc baseType = type.BaseType;
+            DefType baseType = type.BaseType;
             bool hasNonTrivialParent = baseType != null
                 && !baseType.IsWellKnownType(WellKnownType.Object)
                 && !baseType.IsWellKnownType(WellKnownType.ValueType);
 
-            if (hasNonTrivialParent && !IsBlittableType(baseType))
+            // Type is blittable only if parent is also blittable and is not empty.
+            if (hasNonTrivialParent && (!IsBlittableType(baseType) || baseType.InstanceByteCountUnaligned - baseType.InstanceByteAlignment == LayoutInt.Zero))
```

Comment:
The subtraction and compare against Zero doesn't seem right to me. What are the values that you're seeing in these cases? We may need to add a boolean flag like the native VM uses.

Reply:
I just read the code and debugged it with varied types to figure out that the only way in crossgen2 to get the size without padding is when InstanceByteCountUnaligned and InstanceByteAlignment are equal.